core/unix/os_exports.h

```diff
@@ -105,7 +105,11 @@
                              */
 #        define LIB_SEG_TLS DR_REG_TPIDRURO /* libc+loader tls */
 #    endif                                  /* 64/32-bit */
-#endif                                      /* X86/ARM */
+#elif defined(RISCV64)
+/* FIXME-RISCV: Check if this is preserved by kernel and safe to use? */
+#    define SEG_TLS DR_REG_S11
```

Comment:
See above: for a GPR we'd want it option-controlled and not hardcoded, like we do for AArchXX.

Reply:
I'm not sure how to achieve that given that it's a define. Do you suggest refactoring `TLS_REG_ALT` to be a `reg_t` field in `dcontext_t` filled at initialization time?

Reply:
I feel like I need a refresher on what this SEG_TLS is really used for on non-x86: when in the code cache we steal a GPR; when in private lib code we have to use whatever regular lib code uses -- but I think that's LIB_SEG_TLS.  So SEG_TLS is not used at all for AArchXX, and presumably the same would be true for RISC-V?  So maybe just set this to DR_REG_INVALID.

Reply:
You might be right. `DR_REG_INVALID` sounds OK for now.

Reply:
Done.